src/core/main.py

```diff
@@ -57,32 +59,51 @@ def main(
     # Handle any command argument.
     parser = argparse.ArgumentParser()
     parser.add_argument("--debug", default=False, action="store_true")
-    parser.add_argument("--host", default="0.0.0.0", type=str, help="IP")
-    parser.add_argument("--port", default="8000", type=int, help="Port")
+    parser.add_argument("--host", type=str, help="IP")
+    parser.add_argument("--port", type=int, help="Port")
     parser.add_argument("--test", default=False, action="store_true")
     parser.add_argument(
         "--dev", default=False, action="store_true"
     )  # No ops, needed for root-app.
     args, _ = parser.parse_known_args(argsv)
 
+    # Let host argument override config
+    if args.host:
+        logger.info(
+            "Overriding core API hostname from {} to {}".format(
+                config.get("CORE", "hostname"), args.host
+            )
+        )
+        config["CORE"]["hostname"] = args.host
```

Comment:
Instead of overwriting the config object values, I would maybe collect the config values to variable first and if `args` override those. Then there is no chance for the overridden values to be saved to config file accidentally.

Reply:
This would not be possible with my example over other than creating a copy that one doesn't write.  And always explicitly calls save when need be.

Maybe my example is not the way to go then. 🤔

Reply:
I dont think this is a issue since you have to explicitly have to call `write_config(...)` from config package in order to save it. If we ever want the main script to save a configuration file, we can do it when we implement that functionality. As it is now, the ConfigParser object is just a dict to hold values. Overwriting them will not change the config file, unless `write_config(...)` is called. Regardless ill change it as @tomrtk suggest, and not overwrite the configparser object.